test.cpp

```diff
@@ -0,0 +1,28 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "test/catch.hpp"
+#include "main.h"
+
+TEST_CASE("detect set of values  from  specified range within input array of integers") {
+   int sampleArray[] ={2,3, 3, 5, 4, 10, 11, 12};
+   
+   int startRangeValue = 2;
+   int endRangeValue = 5;
+   REQUIRE(getNoOfRValuesInRange(sampleArray,startRangeValue,endRangeValue) == 5);
+   
+   startRangeValue = 10;
+   endRangeValue = 12;
+   REQUIRE(getNoOfRValuesInRange(sampleArray,startRangeValue,endRangeValue) == 3);
+   
+}
+
+
+TEST_CASE("Convert a reading to Ampere with error condition check") {
+  
+   float reading=1146;
+   REQUIRE(ampConverter(reading) == 3);
+   
+   reading=4095;
+   REQUIRE(ampConverter(reading) == -1);
```

Comment:
These tests are good to test the individual functionality. At times, it is efficient to test and catch issues with integration as well, at the unit-test level itself.
Add test to check the integration of ampConverter to getNoOfRValuesInRange: will -1 and 0 be considered consecutive?
The reason this is efficient to test in unit tests - such a test will be very hard to test in a fully integrated (hw+sw) environment